index.md

```diff
@@ -1,16 +1,128 @@
+Setting up the Jupyter kernel is the most straightforward.  If the payload can be represented in a Python virtual environment, `python -m ipykernel install` is all that needs to be done.  If the payload is more complicated (as the DM pipelines stack is), then it turns out that the kernel can be a shell script that does necessary setup and execs a python process as its last action.
+
+The next is setting up a terminal environment.  One way to do this is to set a sentinel environment variable at Jupyterlab launch, and then when the user requests a shell, in the user's `.bashrc` or moral equivalent, test that variable, and if it is set (indicating the user is running inside Jupyterlab), perform the necessary setup (environment activation, running an appropriate setup script, whatever) before yielding control to the user for interactive shell use.
+
+The third is the most difficult.  In Nublado, we have a spawner form.  That form contains a selection of prepulled images, a drop-down box to select an uncached image, a selector to choose container size, and two checkboxes useful for debugging and unwedging stuck environments.  I am not actually sure how generic that is, but I think erring on the side of "everyone will want something like this" is better than insisting it's DM-pipelines-specific.
+
+The purpose of the form in the RSP environment is basically twofold: first, to let the user select how much they need in the way of resources, and second to reflect our release cadence.  We have, effectively, "daily", "weekly", and "release" images and would like some way to present those to the user and update them without human intervention as new images are built.
+
+By having rules about how many of each we use, we can also decide which images to prepull.  Given our use cases, it is vital to prepull images, because they are so large that waiting for a spawn that needs to pull an image from a remote repository becomes quite an unpleasant experience.  If the images are cached to each node (which Nublado does), then for images displayed on the menu (rather than in the drop-down list of all images), the spawn is generally a matter of seconds rather than minutes.  We simply poll the image repository and pull the appropriately-tagged images periodically.  Most of the time, neither the tags nor the content at each tag has changed, and in this case, all the "pull" is the server replying that the image associated with a tag has a particular sha256 checksum, and the client saying "oh, I've got that one, nothing to do, then"; thus while we do many thousands of pulls a day, only a handful actually do much work.
+
+It is not entirely clear that the Rubin practice of builds-on-a-fairly-short-cadence, almost immediately reflected in the spawner, is what all users of Phalanx will want.  However, this form would work just as well with a single type of release and builds updated quarterly, although the mechanisms that populate it will be overcomplicated for the use case.
```

Comment:
Hopefully we could convince anyone with a different approach to use a much simpler version numbering scheme based off of either semver or calver with normal alpha/beta markers. I would be tempted to just implement those and say we don't support anything else, keeping the IMO excessively complicated Rubin scheme around just for us because we're committed to it at this point.